packages/rest/src/body-parsers/helper.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as debugModule from 'debug';
+import {HttpError} from 'http-errors';
+import {Request, Response, RequestBodyParserOptions} from '../types';
+
+import {OptionsJson, OptionsUrlencoded, OptionsText} from 'body-parser';
+const debug = debugModule('loopback:rest:body-parser');
+
+/**
+ * Get the content-type header value from the request
+ * @param req Http request
+ */
+export function getContentType(req: Request): string | undefined {
+  return req.get('content-type');
+}
+
+/**
+ * Express body parser function type
+ */
+export type BodyParserWithCallback = (
+  request: Request,
+  response: Response,
+  callback: (err: HttpError) => void,
+) => void;
+
+/**
+ * Normalize parsing errors as `4xx`
+ * @param err
+ */
+export function normalizeParsingError(err: HttpError) {
+  debug('Cannot parse request body %j', err);
+  if (!err.statusCode || err.statusCode >= 500) {
+    err.statusCode = 400;
+  }
+  return err;
+}
+
+// tslint:disable:no-any
+
+/**
+ * Parse the body asynchronously
+ * @param handle The express middleware handler
+ * @param request Http request
+ */
+export function parseRequest(
+  handle: BodyParserWithCallback,
+  request: Request,
+): Promise<any> {
+  // A hack to fool TypeScript as we don't need `response`
+  const response = ({} as any) as Response;
+  return new Promise<void>((resolve, reject) => {
+    handle(request, response, err => {
+      if (err) {
+        reject(normalizeParsingError(err));
```

Comment:
Do we actually need to normalize the error here? Personally, I'd prefer to rely on `RequestBodyParser` to do the normalization, to ensure we apply the same normalization on errors thrown by custom parsers that don't call `parseRequest` under the hood.

Also, please add a test to verify that when a custom parser throws an error, it is normalized as expected.